src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -120,22 +121,19 @@ private SecurityIdentity login(String username, String password, String mechanis
             return null;
         }
 
-        ServerAuthenticationContext authenticationContext;
-        if(WildFlySecurityManager.isChecking()) {
-            authenticationContext = AccessController.doPrivileged((PrivilegedAction<ServerAuthenticationContext>) () -> securityDomain.createNewAuthenticationContext());
-        } else {
-            authenticationContext = securityDomain.createNewAuthenticationContext();
-        }
-
-        final PasswordGuessEvidence evidence = new PasswordGuessEvidence(password.toCharArray());
+        PasswordGuessEvidence evidence = null;
 
-        try {
+        try (ServerAuthenticationContext authenticationContext = WildFlySecurityManager.isChecking()
+                ? AccessController.doPrivileged((PrivilegedAction<ServerAuthenticationContext>) () -> securityDomain.createNewAuthenticationContext())
+                : securityDomain.createNewAuthenticationContext()) {
```

Comment:
This could be marginally cleaner by creating a utility method to create the new auth context.